exec/totemknet.c

```diff
@@ -1785,36 +1797,50 @@ static int setup_nozzle(void *knet_context)
 
 	res = icmap_get_string(NOZZLE_NAME, &name_str);
 
+	if ((strcmp(name_str, instance->nozzle_name) == 0) &&
+	    (strcmp(ipaddr_str, instance->nozzle_ipaddr) == 0) &&
+	    (strcmp(prefix_str, instance->nozzle_prefix) == 0) &&
+	    (strcmp(macaddr_str, instance->nozzle_macaddr) == 0)) {
```

Comment:
Undefined behavior when NULL is passed. It doesn't crash, but I wouldn't rely on this behavior.